### PR TITLE
feat: add "description", "markdown_description", and "deprecation_message" to resource, provider, and data source schemas

### PR DESCRIPTION
**Pull request by SKalt**
Addresses #81.

These added fields were copied more or less verbatim from the schemas in github.com/hashicorp/terraform-plugin-framework:
- [`resource`](https://github.com/hashicorp/terraform-plugin-framework/blob/c5f89cd/resource/schema/schema.go#L38-L72)
- [`provider`](https://github.com/hashicorp/terraform-plugin-framework/blob/c5f89cd/provider/schema/schema.go#L37-L59)
- [`datasource`](https://github.com/hashicorp/terraform-plugin-framework/blob/c5f89cd/datasource/schema/schema.go#L37-L61)

I chose to include both fields partially to simplify generating `github.com/hashicorp/terraform-plugin-framework` code, but I only need `MarkdownDescription` for my purposes: openapi v3.x.x fields should be interpreted as markdown (see https://spec.openapis.org/oas/v3.0.3#rich-text-formatting, https://spec.openapis.org/oas/v3.1.0#rich-text-formatting).

This change _should_ be a minor change, adding functionality without affecting any existing code.
Notably, this change doesn't check that the `markdown_description` field is valid to avoid any risk of breaking existing code.

I also held off touching any versioning. I'd be happy to handle that if you'd like to bump to 0.2!

### CODE REVIEW

File: spec/v0.1/schema.json [hunk 1]
```diff
@@ -164,6 +173,12 @@
         },
         "blocks": {
           "$ref": "#/$defs/provider_blocks"
+        },
+        "description": {
+          "type": "string"
+        },
+        "markdown_description": {
+          "type": "string"
```
Comment (austinvalle) [inline spec/v0.1/schema.json:181]: Can you add the mapped `deprecation_message` from the Go bindings here?

File: spec/v0.1/schema.json [hunk 2]
```diff
@@ -253,6 +268,12 @@
         },
         "blocks": {
           "$ref": "#/$defs/resource_blocks"
+        },
+        "description": {
+          "type": "string"
+        },
+        "markdown_description": {
+          "type": "string"
```
Comment (austinvalle) [inline spec/v0.1/schema.json:276]: Can you add the mapped `deprecation_message` and `version` from the Go bindings here?

**Review by austinvalle — COMMENTED** [review]
Hi @SKalt 👋🏻, thanks for this PR! I left a couple notes to fully add the properties into the schema. In addition, could you also:
- Add all of these new fields to the [`/spec/v0.1/example.json`](https://github.com/hashicorp/terraform-plugin-codegen-spec/blob/main/spec/v0.1/example.json) and the related test: https://github.com/hashicorp/terraform-plugin-codegen-spec/blob/611f5a8dcd3c63aae9038594e8cd1c53def773e2/spec/specification_test.go#L66


As for release, we (the maintainers) need to think about the version bump, I believe we've made a couple other enhancements that haven't been released yet. We may just bump it to `v0.1.1` since it's still pretty early on for this spec.

**Comment by austinvalle** [comment]
We do also have some documentation that is (frustratingly 😆) not in this repository, if you're so inclined to make an additional PR displaying these new fields: https://github.com/hashicorp/terraform-docs-common/blob/9989eba3efb29bf2a7d060d7f42a6dce993fbd2e/website/docs/plugin/code-generation/specification.mdx#L89-L92

**Comment by SKalt** [comment]
Hi @austinvalle! I pushed the missing sections of the JSON schema and tests (good catch!). I'm not sure when/if I'll get to updating the docs, since I'm getting pulled away for other work.

**Comment by SKalt** [comment]
Hey again @austinvalle, are there any blockers for this PR now that I've added the `deprecation_message`s?

**Review by austinvalle — APPROVED** [review]
@SKalt thanks for making the changes! I did have three small things that I pushed to this PR so I can merge and release it today:
- Changed the `string` values to be `*string` to align with our other description properties
- Removed `Version` go binding: I didn't see this in the schema/example, so I removed it for now. We can come back in a follow-up PR if you need that.
- Added a changelog

**Pull request merged**